### ISSUE TITLE
ExtensionList missing when unsupported version requested for services

### ISSUE THREAD

**Issue by amarmodus** [mojaloop/project#1378]
**Summary**:

When wrong content header is supplied in the request header, following error information is sent and extension list is missing.

```
{
    "errorInformation": {
        "errorCode": "3001",
        "errorDescription": "Unacceptable version requested - Client supplied a protocol version which is not supported by the server"
    }
}
```
**Tasks**:
- [x] Implement code changes in central-services-shared project, and increment version
- [x] Create release for central-services-shared
- [x] Update dependencies for central-event-processor, and create release (needs python fixes for CircleCi)
- [x] Update dependencies for account-lookup-service, and create release
- [x] Update dependencies for central-ledger, and create release
- [x] Update dependencies for ml-api-adapter, and create release
- [x] Update dependencies for quoting-service, and create release
- [x] Update dependencies for simulator, and create release
- [x] Update dependencies for transaction-requests-service, and create release (done already by Rajiv)
- [x] Update dependencies for central-settlement, and create release (needs python fixes for CircleCi)
- [x] Update dependencies for bulk-api-adapter, and create release (needs python fixes for CircleCi)

**PRs**:
- [x] central-services-shared changes:  https://github.com/mojaloop/central-services-shared/pull/231
- [x] central-event-processor changes: https://github.com/mojaloop/central-event-processor/pull/204
- [x] central-event-processor (CircleCi/Python changes): https://github.com/mojaloop/central-event-processor/pull/205
- [x] account-lookup-service changes: https://github.com/mojaloop/account-lookup-service/pull/359
- [x] central-ledger changes: https://github.com/mojaloop/central-ledger/pull/747
- [x] ml-api-adapter changes: https://github.com/mojaloop/ml-api-adapter/pull/443
- [x] quoting-service changes: https://github.com/mojaloop/quoting-service/pull/223
- [x] simulator changes: https://github.com/mojaloop/simulator/pull/196
- [x] transaction-requests-service changes: NOT NEEDED
- [x] central-settlement changes: https://github.com/mojaloop/central-settlement/pull/308
- [x] central-settlement (CircleCi/Python changes): https://github.com/mojaloop/central-settlement/pull/309
- [x] bulk-api-adapter changes: https://github.com/mojaloop/bulk-api-adapter/pull/30
- [x] bulk-api-adapter (CircleCi/Python changes): https://github.com/mojaloop/bulk-api-adapter/pull/31

**Severity**:
Medium

**Priority**:
Low

**Expected Behavior**
```
{ 
     "errorInformation": { 
               "errorCode": "3001", 
                "errorDescription": "The Client requested an unsupported version, see extension list for supported version(s).", 
                  "extensionList": [ 
                           { "key": "1", "value": "0" }, 
                           { "key": "2", "value": "1" }, 
                           { "key": "4", "value": "2" } 
                    ]
         } 
} 
```

**Steps to Reproduce**
  ```
curl --location --request GET 'http://dev2-transaction-requests-service.mojaloop.live/authorizations/43638814-1a02-44eb-9e93-b6dd295799d7?authenticationType=OTP&retriesLeft=2&amount=102&currency=USD' \
--header 'Accept: application/vnd.interoperability.authorizations+json;version=1' \
--header 'Content-Type: application/vnd.interoperability.authorizations+json;version=4.0' \
--header 'Date: Mon, 18 May 2020 15:01:49 GMT' \
--header 'fspiop-source: payerfsp' \
--header 'fspiop-destination: payeefsp' \
--header 'fspiop-signature: {"signature":"iU4GBXSfY8twZMj1zXX1CTe3LDO8Zvgui53icrriBxCUF_wltQmnjgWLWI4ZUEueVeOeTbDPBZazpBWYvBYpl5WJSUoXi14nVlangcsmu2vYkQUPmHtjOW-yb2ng6_aPfwd7oHLWrWzcsjTF-S4dW7GZRPHEbY_qCOhEwmmMOnE1FWF1OLvP0dM0r4y7FlnrZNhmuVIFhk_pMbEC44rtQmMFv4pm4EVGqmIm3eyXz0GkX8q_O1kGBoyIeV_P6RRcZ0nL6YUVMhPFSLJo6CIhL2zPm54Qdl2nVzDFWn_shVyV0Cl5vpcMJxJ--O_Zcbmpv6lxqDdygTC782Ob3CNMvg\",\"protectedHeader\":\"eyJhbGciOiJSUzI1NiIsIkZTUElPUC1VUkkiOiIvdHJhbnNmZXJzIiwiRlNQSU9QLUhUVFAtTWV0aG9kIjoiUE9TVCIsIkZTUElPUC1Tb3VyY2UiOiJPTUwiLCJGU1BJT1AtRGVzdGluYXRpb24iOiJNVE5Nb2JpbGVNb25leSIsIkRhdGUiOiIifQ"}' \
--header 'fspiop-uri: /transactionRequests' \
--header 'fspiop-http-method: GET'
```

**Specifications**
  - Component (if known): Error framework
  - Version:v10.1.0
  - Platform: 
  - Subsystem:
  - Type of testing: Manual postman test
  - Bug found/raised by: @amarmodus 

**Notes**:

- Severity when opened:
- Priority when opened:

**Comment by aenns** [comment]
@amarmodus @elnyry-sam-k 
Where do the values for the extension list come from?

**Comment by elnyry-sam-k** [comment]
@aenns - these have to set in a central place somewhere, sort of like [sdk-standard-components](https://github.com/mojaloop/sdk-standard-components). The valid values right now are "1.0" (key-1, value-0) and "1.1"(key-1, value-1); These represent the versions of the FSPIOP Spec that are supported by an implementation.. (Currently we have v1.1 and v1.0 supported, though v1.1 is still in progress, one issue is left)

**Comment by aenns** [comment]
@elnyry-sam-k 
I am able to replicate the error. This is the code that throws the exception:

if (!supportedProtocolVersions.includes(contentType.version)) {
        throw createFSPIOPError(
          Enums.FSPIOPErrorCodes.UNACCEPTABLE_VERSION,
          errorMessages.SUPPLIED_VERSION_NOT_SUPPORTED
        )
      }

The supported versions are hard-coded in a variable (in the central-services-shared project) and currently only support "1" and "1.0". So, if we tried a request with a different version (such as "1.1"), it will throw this error (unless I'm looking in the wrong place, but I tested this as well with "1.1" and it caused the error to occur). My question here is whether I add "1.1" to the supported versions, or I leave it as is, and DO NOT include "1.1" in the extension list?
